extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -379,41 +416,63 @@ class AmpLightbox extends AMP.BaseElement {
     owners.scheduleResume(this.element, container);
     this.triggerEvent_(LightboxEvents.OPEN, trust);
 
+    this.focusInModal_();
+
     this.getHistory_()
       .push(this.close.bind(this))
-      .then((historyId) => {
+      .then(historyId => {
         this.historyId_ = historyId;
       });
 
     this.active_ = true;
   }
 
-  /** @private */
+  /**
+   * Renders a top bar with close button if the attribute close-button is set
+   * Used for ad if no close button is set.
+   * @private
+   */
   maybeRenderCloseButtonHeader_() {
     const {element} = this;
-
     if (element.getAttribute('close-button') == null) {
       return;
     }
 
-    const header = renderCloseButtonHeader(element);
+    this.closeButtonHeader_ = renderCloseButtonHeader(element);
+    element.insertBefore(this.closeButtonHeader_, this.container_);
 
-    this.closeButtonHeader_ = header;
-
-    // Click gesture is high trust.
-    listenOnce(header, 'click', () => this.close(ActionTrust.HIGH));
+    this.closeButtonHeader_.addEventListener('click', () => {
+      // Click gesture is high trust.
+      this.close(ActionTrust.HIGH);
+    });
 
-    element.insertBefore(header, this.container_);
+    this.boundCloseOnEnter_ = /** @type {?function(this:AmpLightbox, Event)} */ (this.closeOnEnter_.bind(
+      this
+    ));
+    this.closeButtonHeader_.addEventListener(
```

Comment:
You think I should addEventListener on open then remove on close instead of just addint it to initialize ?
I've seen both solutions been used so I thought its ok to use on initialize